Rabail branch

rabail-aamir opened the pull request:
# Description
This pull request adds a new demo project zip file ArrowGamecsharp.zip to the DemoProjects/ folder and removes the previously created (now empty) demo/ folder. This update helps maintain a cleaner project structure and ensures the demo file is correctly placed and accessible.
The demo zip contains a C# SplashKit game showcasing basic player movement using arrow keys. It can serve as a sample project for contributors and learners


## How Has This Been Tested?

- Verified that the file ArrowGamecsharp.zip is properly placed in DemoProjects/ folder
- Checked file structure in local repo after moving and removing the old folder
- Confirmed that demo/ no longer exists
- Validated file upload and visibility on GitHub in the correct folder

## Testing Checklist

- [✓ ] Tested in latest Chrome
- [✓ ] Tested in latest Safari
- [✓ ] Tested in latest Firefox

_**Since this is a static file addition, browser testing is only relevant for download/accessibility.**_

## Checklist

- [✓] My code follows the style guidelines of this project
- [✓ ] I have performed a self-review of my own code
- [✓] I have commented my code in hard-to-understand areas
- [✓ ] I have made corresponding changes to the documentation
- [✓ ] My changes generate no new warnings
- [✓ ] I have requested a review from ... on the Pull Request

rabail-aamir committed:
Description of the changes

rabail-aamir committed:
Add ArrowGamecsharp.zip to demo folder

rabail-aamir committed:
Add ArrowGamecsharp.zip to DemoProjects folder

rabail-aamir committed:
Remove empty demo folder

rabail-aamir committed:
Add Arrow Game (C#) demo with clean zip, thumbnail, and demos.json entry

rabail-aamir committed:
Remove .vscode/settings.json from tracking and add to .gitignore

Oliver-Quail commented on .vscode/settings.json:
Hi Rabail,
Please can you delete this file as it shouldn't be uploaded to the repo. I will look into why this .gitignore is not ignoring this

Oliver-Quail reviewed (changes requested):
Hi Rabail,
I had a look over your PR and it only needs a few minor changes.
Firstly please add your demo to the ```DemoProjects/metadata/demos.json``` file. For more details, please see the guide in [this PR](https://github.com/thoth-tech/SplashkitOnline/pull/131)
Also you have uploaded some local files that are not needed in SplashKit Online. Please delete these file in the highlighted image and re zip and upload the project
<img width="338" height="682" alt="review" src="https://github.com/user-attachments/assets/d5c23ce3-1ced-48cc-85d7-2ef6b1dd6789" />
Finally, when testing their was an issue with running the game. **This is an issue with the SK Online C# not being fully implemented**. I will look into fixing this as it is a complex issue, please don't worry about this.

Overall good first PR :).
Please let me know when you've fixed the above issues and I will be happy to approve :)

rabail-aamir commented:
Hi Oliver,
I’ve made the following updates in this branch:
- Added ArrowGamecsharp.zip to DemoProjects/ with its thumbnail and updated demos.json metadata accordingly.
- Removed .vscode/settings.json from the repository and added .vscode/ to .gitignore so it won’t be tracked in the future.

Oliver-Quail reviewed (approved):
I approve this pr

liammcknightt reviewed (approved):
The addition of this demo program is as per the documentation in PR #131.

ZIP folder is in the DemoProjects folder.
There is a screenshot of the running program.
Has been added to demos.json in a new array for C# programs.

As per the previous review on this PR #129 I was not able to test the game locally as C# is not properly implemented in SK Online yet.

liammcknightt commented:
I have moved this to Mentor Review
